### PR TITLE
[MNT-23379] Set site manager permission if node belongs to a site and inheritance flag is disabled

### PR DESCRIPTION
**Pull request by tiagosalvado10**

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 1]
```diff
@@ -2763,6 +2766,38 @@ public Node updateContent(String fileNodeId, BasicContentInfo contentInfo, Input
...
+                {
+                    SiteInfo containingSite = siteService.getSite(nodeRef);
+
+                    if (containingSite != null)
+                    {
+                        String thisSiteGroupPrefix = siteService.getSiteGroup(containingSite.getShortName());
+                        final String siteManagerAuthority = thisSiteGroupPrefix + "_" + SiteModel.SITE_MANAGER;
+                        AuthenticationUtil.runAs(new AuthenticationUtil.RunAsWork<Void>()
+                        {
+                            public Void doWork() throws Exception
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java:2785]: The `runAs` can be replaced by a lambda function if you want.

**Review by tpage-alfresco — APPROVED** [review]
LGTM

File: remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java (hunk 1 above)
Comment (tiagosalvado10) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Replaced the `runAs` call as suggested.

**Review by tpage-alfresco — APPROVED** [review]
Looks even better :+1: :+1:

**Review by jottley — APPROVED** [review]

**Pull request merged**